core/domain/query_jobs_one_off.py

```diff
@@ -0,0 +1,143 @@
+
+            for exp in edited_explorations:
+                days_since_updated = (
+                    datetime.datetime.utcnow() - exp.last_updated).days
+                if days_since_updated <= days:
+                    edited_exploration_in_n_days = True
+                    break
+
+            user_qualified = (
+                (created_exploration_in_n_days or edited_exploration_in_n_days)
+                and user_qualified)
+
+        if (query_parameters.created_more_than_n_exploration is not None and
+                user_qualified):
+            exploration_count = query_parameters.created_more_than_n_exploration
+            user_contributions = user_models.UserContributionsModel.get(user_id)
+            created_explorations = exp_services.get_multiple_explorations_by_id(
+                user_contributions.created_exploration_ids)
+
+            user_qualified = (
+                (len(created_explorations) >= exploration_count) and
+                user_qualified)
+
+        if (query_parameters.created_less_than_n_exploration is not None and
+                user_qualified):
+            exploration_count = query_parameters.created_less_than_n_exploration
+            user_contributions = user_models.UserContributionsModel.get(user_id)
+            created_explorations = exp_services.get_multiple_explorations_by_id(
+                user_contributions.created_exploration_ids)
+
+            user_qualified = (
+                (len(created_explorations) <= exploration_count) and
+                user_qualified)
+
+        if (query_parameters.edited_more_than_n_exploration is not None and
+                user_qualified):
+            exploration_count = query_parameters.edited_more_than_n_exploration
+            user_contributions = user_models.UserContributionsModel.get(user_id)
+            edited_explorations = exp_services.get_multiple_explorations_by_id(
+                user_contributions.edited_exploration_ids)
+
+            user_qualified = (
+                (len(edited_explorations) >= exploration_count) and
+                user_qualified)
+
+        if (query_parameters.edited_less_than_n_exploration is not None and
+                user_qualified):
+            exploration_count = query_parameters.edited_less_than_n_exploration
+            user_contributions = user_models.UserContributionsModel.get(user_id)
+            edited_explorations = exp_services.get_multiple_explorations_by_id(
+                user_contributions.edited_exploration_ids)
+
+            user_qualified = (
+                (len(edited_explorations) <= exploration_count) and
+                user_qualified)
+
+        if user_qualified:
+            yield (query_id, user_id)
+
+    @staticmethod
+    def reduce(query_model_id, stringified_user_ids):
+        query = job_models.QueryModel.get(query_model_id)
+        user_ids = [ast.literal_eval(v) for v in stringified_user_ids]
+        new_user_ids = set(query.user_ids + user_ids)
```

Comment:
Yeah. Generally speaking there won't be any ids in model. I did it because I wasn't sure about how shuffled stage of MR job works. From what I read, all values for same key are gathered in single list and then passed over to reducer. But I was not sure about it. 

If you are sure the I will remove it. Thanks